lib/rom/processor/transproc.rb

```diff
@@ -99,6 +99,8 @@ def visit(attribute)
       def visit_attribute(attribute)
         if attribute.typed?
           t(:map_value, attribute.name, t(:"to_#{attribute.type}"))
+        elsif attribute.meta[:coercer]
+          t(:map_value, attribute.name, attribute.meta[:coercer])
         end
```

Comment:
I believe `:coercer` should take precedence here and we should clearly document it, maybe even raise when somebody tries to define `type` and at the same time provide a coercion block.